src/main/java/org/rumbledb/compiler/StaticContextVisitor.java

```diff
@@ -226,9 +226,7 @@ public StaticContext visitCountClause(CountClause expression, StaticContext argu
     private StaticContext visitFlowrVarDeclaration(FlworVarDecl expression, StaticContext argument) {
         StaticContext result = new StaticContext(argument);
         // TODO for now we only suppot as/default, no inference, flags
-        SequenceType type = expression.getSequenceType() == null
-            ? SequenceType.emptySequence
-            : expression.getSequenceType();
+        SequenceType type = expression.getSequenceType();
```

Comment:
we can even remove this line and just call expression.getSequenceType() at line 232

Reply:
Good point, done